src/js/program/ad-program-controller.js

```diff
@@ -114,6 +114,9 @@ export default class AdProgramController extends ProgramController {
             event.oldstate = model.get(PLAYER_STATE);
             adMediaModelContext.set('mediaState', event.newstate);
         });
+        provider.on(NATIVE_FULLSCREEN, (event) => {
+            this._nativeFullscreenHandler(Object.assign({}, event, { type: NATIVE_FULLSCREEN }));
```

Comment:
That looks like it's doing too much.

How about this:
```js
provider.on(NATIVE_FULLSCREEN, this._nativeFullscreenHandler, this);

// Then, since you already know the type of event you are triggering:
_nativeFullscreenHandler(evt) {
    this.model.trigger(NATIVE_FULLSCREEN, evt);
    this.trigger(FULLSCREEN, {
        fullscreen: evt.jwstate
    });
}
```